src/main/java/frc/robot/commands/TeleopDrive.java

```diff
@@ -1 +1,161 @@
+/* the project.                                                               */
+/*----------------------------------------------------------------------------*/
 
+package frc.robot.commands;
+
+import edu.wpi.first.wpilibj.Joystick;
+import edu.wpi.first.wpilibj2.command.CommandBase;
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
+import frc.robot.subsystems.Drivetrain;
+
+public class TeleopDrive extends CommandBase {
+  Drivetrain dt;
+  Joystick leftJoy, rightJoy;
+
+  double prevSpeed = 0, prevLeft = 0, prevRight = 0;
+
+  double outreachSpeed = 0.3;
+
+  /**
+   * Handles all the teleoperated driving functionality
+   * 
+   * @param dt the Drivetrain object to use, passing it in is useful for testing
+   *           purposes
+   */
+  public TeleopDrive(Drivetrain dt, Joystick leftJoy, Joystick rightJoy) {
+    this.dt = dt;
+    this.leftJoy = leftJoy;
+    this.rightJoy = rightJoy;
+    addRequirements(dt);
+    
+    SmartDashboard.putBoolean("Characterized Drive", false);
+    
+  }
+
+    @Override
+    public void execute() {
+        if (SmartDashboard.getBoolean("Arcade Drive", true)) {
+            arcadeDrive();
+        } else {
+            tankDrive();
+        }
+    }
+
+  private void arcadeDrive() {
+    double speed = -leftJoy.getY();
+    double rot = rightJoy.getX();
+
+    // System.out.println("Speed: " + speed + ", Rotation: " + rot);
+    SmartDashboard.putNumber("input speed", speed);
+    SmartDashboard.putNumber("input rot", rot);
+
+    if (SmartDashboard.getBoolean("Square Joysticks", true)) {
+      speed = Math.copySign(speed * speed, speed);
+      rot = Math.copySign(rot * rot, rot);
+    }
+
+    if (SmartDashboard.getBoolean("Slow Left", false)) {
+      speed *= SmartDashboard.getNumber("Speed Slow Ratio", 0.5);
+    }
+    if (SmartDashboard.getBoolean("Slow Right", false)) {
+      rot *= SmartDashboard.getNumber("Rotation Slow Ratio", 0.35);
+    }
+
+    if (SmartDashboard.getBoolean("Gradual Drive", true)) {
+      double dV = SmartDashboard.getNumber("Gradual Drive Max dV", 0.04);
+      if (Math.abs(speed - prevSpeed) > dV) {
+        speed = prevSpeed + dV * Math.signum(speed - prevSpeed);
+      }
+    }
+    prevSpeed = speed;
+
+    double left, right;
+
+    // double maxInput = Math.copySign(Math.max(Math.abs(speed), Math.abs(rot)),
+    // speed);
+    // copySign is returning incorrect signs in operation but not tests
+    double maxInput = Math.max(Math.abs(speed), Math.abs(rot));
+
+    if (speed >= 0.0) {
+      // First quadrant, else second quadrant
+      if (rot >= 0.0) {
+        left = maxInput;
+        right = speed - rot;
+      } else {
+        left = speed + rot;
+        right = maxInput;
+      }
+    } else {
+      // Third quadrant, else fourth quadrant
+      maxInput *= -1;
+      if (rot >= 0.0) {
+        left = speed + rot;
+        right = maxInput;
+      } else {
+        left = maxInput;
+        right = speed - rot;
+      }
```

Comment:
instead of this, you can just use a [DifferentialDrive](https://first.wpi.edu/FRC/roborio/release/docs/java/edu/wpi/first/wpilibj/drive/DifferentialDrive.html) and the tankDrive and arcadeDrive methods which just take the joystick inputs